### DISCUSSION TITLE
error uploading an image with livewire

### DISCUSSION THREAD

**Discussion by ritechoice23** [livewire/livewire#4998] [1. Help]
I have been trying to upload an image with livewire, whenever i select an image i always get this error. 
**The thumbnail failed to upload.**
below is my code

blade 

 <form wire:submit.prevent='update'>
            @if ($thumbnail)
            <img src="{{ $thumbnail->temporaryUrl() }}">
            @endif
            <div class="mt-3">
                <div class="flex items-center justify-center w-full">
                    <label id="thumbnail"
                        class="flex flex-col items-center justify-center w-full h-64 border-2 border-gray-300 border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600">
                        <div class="flex flex-col items-center justify-center pt-5 pb-6">
                            <svg class="w-10 h-10 mb-3 text-gray-400" fill="none" stroke="currentColor"
                                viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                                    d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12">
                                </path>
                            </svg>
                            <p class="mb-2 text-sm text-gray-500 dark:text-gray-400"><span class="font-semibold">Click
                                    to upload</span>
                                or drag and drop</p>
                            <p class="text-xs text-gray-500 dark:text-gray-400">SVG, PNG, JPG or GIF (MAX. 800x400px)
                            </p>
                        </div>
                        <input id="thumbnail" type="file" wire:model='thumbnail' class="hidden">
                    </label>

                    @error('thumbnail')
                    <div>{{ $message }}</div>
                    @enderror
                </div>
            </div>

            <div class="mt-3">
                <x-button type='submit' spinner='update' primary label="Save" />
            </div>

        </form>

i have defined the prop in the class

**Comment by joshhanley** [comment]
@ritechoice23 please fix the formatting of your post, it seems there is code missing.

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Thanks!
